src/accessors/ISubscriptionRead.ts

```diff
@@ -0,0 +1,15 @@
+import { ISubscription } from '../subscriptions/ISubscription';
+
+/**
+ * This accessor provides methods for accessing
+ * subscriptions in a read-only-fashion.
+ */
+export interface ISubscriptionRead {
+    /**
+     * Gets a subscription by a room id.
+     *
+     * @param roomId the id of the room
+     * @returns the subscription
+     */
+    getByRoomId(roomId: string): Promise<IterableIterator<ISubscription>>;
```

Comment:
Can you help me understand why you're returning an `IterableIterator` here when you're only returning one? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols

Reply:
the comment is wrong, it's supposed to return an Iterable of ISubscription

Reply:
Okay, so maybe the one above it should read `Gets all of the subscriptions which are for the provided room.` as the `gets a subscription` is a little misleading as well.

Reply:
Perfect! Thanks!